sympy/solvers/ode/tests/test_systems.py

```diff
@@ -781,11 +952,129 @@ def test_linear_3eq_order1_type4_slow():
+            c2**2*r2**2)/(2*c1*c2*r1*r2))/sqrt(c1**2*r1**2 + 2*c1**2*r1*r2 + c1**2*r2**2 - 2*c1*c2*r1*r2 +
+            2*c1*c2*r2**2 + c2**2*r2**2), t)/(c1*r1 + c1*r2 - c2*r2 - sqrt(c1**2*r1**2 + 2*c1**2*r1*r2 +
+            c1**2*r2**2 - 2*c1*c2*r1*r2 + 2*c1*c2*r2**2 + c2**2*r2**2))), Eq(x2(t), C1*exp(-t/(2*c2*r2) -
+            t/(2*c2*r1) - t/(2*c1*r1) - t*sqrt(c1**2*r1**2 + 2*c1**2*r1*r2 + c1**2*r2**2 - 2*c1*c2*r1*r2 +
+            2*c1*c2*r2**2 + c2**2*r2**2)/(2*c1*c2*r1*r2)) + C2*exp(-t/(2*c2*r2) - t/(2*c2*r1) - t/(2*c1*r1) +
+            t*sqrt(c1**2*r1**2 + 2*c1**2*r1*r2 + c1**2*r2**2 - 2*c1*c2*r1*r2 + 2*c1*c2*r2**2 +
+            c2**2*r2**2)/(2*c1*c2*r1*r2)) + exp(-t/(2*c2*r2) - t/(2*c2*r1) - t/(2*c1*r1) - t*sqrt(c1**2*r1**2 +
+            2*c1**2*r1*r2 + c1**2*r2**2 - 2*c1*c2*r1*r2 + 2*c1*c2*r2**2 +
+            c2**2*r2**2)/(2*c1*c2*r1*r2))*Integral(-i*r2*exp(t/(2*c2*r2) + t/(2*c2*r1) + t/(2*c1*r1) +
+            t*sqrt(c1**2*r1**2 + 2*c1**2*r1*r2 + c1**2*r2**2 - 2*c1*c2*r1*r2 + 2*c1*c2*r2**2 +
+            c2**2*r2**2)/(2*c1*c2*r1*r2))/sqrt(c1**2*r1**2 + 2*c1**2*r1*r2 + c1**2*r2**2 - 2*c1*c2*r1*r2 +
+            2*c1*c2*r2**2 + c2**2*r2**2), t) + exp(-t/(2*c2*r2) - t/(2*c2*r1) - t/(2*c1*r1) + t*sqrt(c1**2*r1**2
+            + 2*c1**2*r1*r2 + c1**2*r2**2 - 2*c1*c2*r1*r2 + 2*c1*c2*r2**2 +
+            c2**2*r2**2)/(2*c1*c2*r1*r2))*Integral(i*r2*exp(t/(2*c2*r2) + t/(2*c2*r1) + t/(2*c1*r1) -
+            t*sqrt(c1**2*r1**2 + 2*c1**2*r1*r2 + c1**2*r2**2 - 2*c1*c2*r1*r2 + 2*c1*c2*r2**2 +
+            c2**2*r2**2)/(2*c1*c2*r1*r2))/sqrt(c1**2*r1**2 + 2*c1**2*r1*r2 + c1**2*r2**2 - 2*c1*c2*r1*r2 +
+            2*c1*c2*r2**2 + c2**2*r2**2), t))]
+    assert dsolve(eq) == sol
+    assert checksysodesol(eq, sol) == (True, [0, 0])
+
+
+# A very big solution is obtained for this
+# test case. To be simplified in future.
+def test_linear_new_order1_type2_de_lorentz():
+    m = Symbol("m", real=True)
+    q = Symbol("q", real=True)
+    t = Symbol("t", real=True)
+
+    e1, e2, e3 = symbols("e1:4", real=True)
+    b1, b2, b3 = symbols("b1:4", real=True)
+    v1, v2, v3 = symbols("v1:4", cls=Function, real=True)
+
+    eqs = [
+        -e1 * q + m * Derivative(v1(t), t) - q * (-b2 * v3(t) + b3 * v2(t)),
+        -e2 * q + m * Derivative(v2(t), t) - q * (b1 * v3(t) - b3 * v1(t)),
+        -e3 * q + m * Derivative(v3(t), t) - q * (-b1 * v2(t) + b2 * v1(t))
+    ]
+
+    # NOTE: With the current state of the solvers, the solution for this
+    # system is too complicated to be compared with the dsolve's output
+    dsolve(eqs)
+
+
 @slow
-def test_linear_3eq_order1_type4_skip():
+def test_linear_new_order1_type2_de_lorentz_fail():
+    if ON_TRAVIS:
+        skip("Too slow for travis.")
+
+    m = Symbol("m", real=True)
+    q = Symbol("q", real=True)
+    t = Symbol("t", real=True)
+
+    e1, e2, e3 = symbols("e1:4", real=True)
+    b1, b2, b3 = symbols("b1:4", real=True)
+    v1, v2, v3 = symbols("v1:4", cls=Function, real=True)
```

Comment:
Rather than duplicating this code here and above I think that it would be better to make a function that returns the system and expected solution and then call that in each of the two test functions here.